packages/blocks/src/root-block/edgeless/components/color-picker/button.ts

```diff
@@ -29,98 +54,118 @@ export class EdgelessColorPickerButton extends WithDisposable(LitElement) {
-                    })}
-                  ></div>
-                </div>
-              </edgeless-color-panel>
+                  <edgeless-color-custom-button
+                    slot="custom"
+                    style=${`--c: ${this.customButtonColor}`}
+                    .active=${!this.isCSSVariable}
+                    @click=${this.#switchToCustomTab}
+                  ></edgeless-color-custom-button>
+                </edgeless-color-panel>
+              </div>
             `,
           ],
           [
             'custom',
-            () => {
-              const rgb: Rgb = { r: 0, g: 0, b: 0 };
-              const value = window
-                .getComputedStyle(this)
-                .getPropertyValue(this.color);
-              if (value.startsWith('rgb')) {
-                const [a, b, c] = value
-                  .replace(/^rgba?/, '')
-                  .replace(/\(|\)/, '')
-                  .split(',')
-                  .map(s => parseFloat(s.trim()));
-                rgb.r = a;
-                rgb.g = b;
-                rgb.b = c;
-              }
-              const color = rgb;
-              console.log(color);
-
-              return html`
-                <edgeless-color-picker
-                  .color=${color}
-                  .navType=${'colors'}
-                ></edgeless-color-picker>
-              `;
-            },
+            () => html`
+              <edgeless-color-picker
+                .slots=${this.slots}
+                .colors=${{
+                  type: this.colorType,
+                  modes: this.colors.map(
+                    preprocessColor(window.getComputedStyle(this))
+                  ),
+                }}
+              ></edgeless-color-picker>
+            `,
           ],
         ])}
       </editor-menu-button>
     `;
   }
 
+  get colorWithoutAlpha() {
+    return this.isCSSVariable ? this.color : keepColor(this.color);
+  }
+
+  get customButtonColor() {
+    return this.isCSSVariable ? 'transparent' : keepColor(this.color);
+  }
+
+  get isCSSVariable() {
+    return this.color.startsWith('--');
+  }
+
+  get tabContentPadding() {
+    return `${this.tabType === 'custom' ? 0 : 8}px`;
+  }
+
   @property()
   accessor color!: string;
 
-  @property({ type: Array })
-  accessor colors: string[] = [];
+  @property()
+  accessor colorType: PickColorType = 'palette';
+
+  @property({ attribute: false })
+  accessor colors: { type: ModeType; value: string }[] = [];
+
+  @property({ attribute: false })
+  accessor hollowCircle: boolean = false;
+
+  @property()
+  accessor isText!: boolean;
 
   @property()
   accessor label!: string;
 
   @query('editor-menu-button')
   accessor menuButton!: EditorMenuButton;
 
-  @property()
+  @property({ attribute: false })
+  accessor palettes: string[] = [];
+
+  accessor slots = {
```

Comment:
1. These slots have never been disposed.
2. I think we should get rid of `slot` since we're migrating to signal.